src/UCTNode.cpp

```diff
@@ -315,8 +315,10 @@ UCTNode* UCTNode::uct_select_child(Color color, bool is_root) {
     // Estimated eval for unknown nodes = original parent NN eval - reduction
     // Or curent parent eval - reduction if dynamic_eval is enabled.
     auto fpu_eval = (cfg_fpu_dynamic_eval ? get_eval(color) : net_eval) - fpu_reduction;
+    auto bestit = m_children.begin();
 
-    for (const auto& child : m_children) {
+    for (auto it = m_children.begin(); it != m_children.end(); it++) {
```

Comment:
You don't need to change this to a manual loop just to have a pointer to the best child. In fact, it's not clear to me why you insist on having to have an iterator (which is really a pointer)?